packages/core/lib/commands/test/run.js

```diff
@@ -7,32 +7,16 @@ const parseCommandLineFlags = options => {
   const reporter = options.reporter || options.r;
 
   /**
-   * This if-else condition is explicitly written to avoid the overlapping of
-   * the config by the default mocha reporter type when user specifies a mocha reporter type
-   * in the config and doesn't specify it as the command line argument.
-   * If the reporter is returned as undefined, it ignores the specification of any reporter type in the
-   * config and displays the default mocha reporter "spec", as opposed to reporter completely being absent
-   * which results in checking for the reporter type specified in the config.
+   * If the reporter type is undefined in the command line, it results in checking and returning the
+   * reporter type in config if exists, otherwise returns the default mocha reporter type "spec".
```

Comment:
This comment is not correct. Mocha defaults to "spec", we don't specify it anywhere. What is the point of this comment exactly? Does it mean to say something like, "Truffle gives precedence to a reporter value passed on the command line, then to a value specified in the config, and if neither exist then we don't pass it to mocha and it uses its default one"?

Reply:
> This comment is not correct. Mocha defaults to "spec", we don't specify it anywhere. What is the point of this comment exactly? Does it mean to say something like, "Truffle gives precedence to a reporter value passed on the command line, then to a value specified in the config, and if neither exist then we don't pass it to mocha and it uses its default one"?

This is exactly what it supposed to mean. I agree that this comment needs to be changed.